src/Components/Facility/FacilityCreate.tsx

```diff
@@ -844,6 +873,43 @@ export const FacilityCreate = (props: FacilityProps) => {
                     required
                     types={["mobile", "landline"]}
                   />
+                  {facilityId && (
+                    <MultiSelectFormField
+                      {...field("hubs")}
+                      placeholder={t("hubs")}
+                      className={facilitiesQuery.loading ? "animate-pulse" : ""}
+                      disabled={facilitiesQuery.loading || hubsQuery.loading}
+                      options={
+                        facilities?.results.filter(
+                          (f) => f.id !== facilityId,
+                        ) || []
+                      }
+                      optionLabel={(o) => o.name}
+                      optionValue={(o) => o.id}
+                      value={hubFacilities}
+                      onChange={async (event) => {
+                        if (event.value.length > hubFacilities.length) {
+                          await createHub(
+                            event.value[event.value.length - 1] || "",
+                          );
+                        } else if (event.value.length < hubFacilities.length) {
+                          console.log(
+                            hubFacilities.find((x) => !event.value.includes(x)),
+                          );
+                          await deleteHub(
+                            hubsQuery.data?.results.find(
+                              (r) =>
+                                r.hub.id ===
+                                (hubFacilities.find(
+                                  (x) => !event.value.includes(x),
+                                ) || ""),
+                            )?.external_id || "",
+                          );
+                        }
+                        setHubFacilities(event.value as string[]);
```

Comment:
```suggestion
                        hubsQuery.refetch();
```

Reply:
I can do that but this will induce latency between input and showing it to the user right? For user experience I suggest we use the state. Or do we want it this way?

Reply:
Oh yes, that makes sense. Having the state is fine then.